org-roam.el

```diff
@@ -587,6 +591,42 @@ This needs to be quick/infrequent, because this is run at
     (call-process org-roam-graphviz-executable nil 0 nil temp-dot "-Tsvg" "-o" temp-graph)
     (call-process org-roam-graph-viewer nil 0 nil temp-graph)))
 
+(defun org-roam--rename-file-links (file new-file &rest args)
+  "Rename backlinks of FILE to refer to NEW-FILE."
+  (when (and (not (auto-save-file-name-p file))
+             (not (auto-save-file-name-p new-file))
+             (org-roam--org-roam-file-p new-file))
+    (org-roam--ensure-cache-built)
+
+    (let* ((files (gethash file org-roam-backward-links-cache nil))
+           (path (org-roam--relative-path file))
+           (new-path (org-roam--relative-path new-file))
+
+           (slug (org-roam--get-title-or-slug file))
+           (old-title (format org-roam-link-title-format slug))
+
+           (new-slug (or (org-roam--get-title-from-cache (file-truename file))
+                         (org-roam--get-title-or-slug new-path)))
+           (new-title (format org-roam-link-title-format new-slug)))
+      (when files
+        (save-excursion
```

Comment:
Move this `save-excursion` inside the function. Just outside of the replace-regexp. It is not doing anything out here.